src/db/StatusRepository.ts

```diff
@@ -0,0 +1,34 @@
+import { Pool } from 'mysql2/promise'
+
+type Status = {
+    account_id: number
+    updated_at: Date
+    status: string[]
+}
+
+class StatusRepository {
+    pool: Pool
+
+    constructor(pool: Pool) {
+        this.pool = pool
+    }
+
+    // Reads from the `updates` table and returns the latest status
+    // updates for the given accountId and endpoint.
+    // (Updates are stored in JSON format)
+    async getStatus(accountId: number, endpoint: string): Promise<Status> {
+        const query = `SELECT * FROM updates WHERE account_id = ? AND endpoint = ? ORDER BY updated_at DESC LIMIT 1`
```

Comment:
`ORDER BY timestamp` and you have to add a WHERE clause using the timestamp to be able to make use of the index and do not trigger a full table scan. I would recommend to specify timestamp > DATE_SUB(NOW(), INTERVAL 7 DAY) or something like this

Reply:
done